crypto/dh/dh_key.c

```diff
@@ -375,20 +375,17 @@ int ossl_dh_buf2key(DH *dh, const unsigned char *buf, size_t len)
     int err_reason = DH_R_BN_ERROR;
     BIGNUM *pubkey = NULL;
     const BIGNUM *p;
-    size_t p_size;
+    int ret;
 
     if ((pubkey = BN_bin2bn(buf, len, NULL)) == NULL)
         goto err;
     DH_get0_pqg(dh, &p, NULL, NULL);
-    if (p == NULL || (p_size = BN_num_bytes(p)) == 0) {
+    if (p == NULL || BN_num_bytes(p) == 0) {
         err_reason = DH_R_NO_PARAMETERS_SET;
         goto err;
     }
-    /*
-     * As per Section 4.2.8.1 of RFC 8446 fail if DHE's
-     * public key is of size not equal to size of p
-     */
```

Comment:
The referenced TLS RFC still seems relevant but perhaps not here? Should we be checking whether p is padded correctly in libssl instead? The RFC doesn't actually say that peers should validate this, but it does say the endpoints emitting these keys should send them padded.

Reply:
I would not bother validating this. I cannot see any issue coming from this missing size validation.

Reply:
Checking that the key share is padded to the proper encoded length is part of being a good citizen in the broader ecosystem.  If enough implementations are lax in checking, then implementations that are lax in sending can escape into the wild and force a burden of compatibility (accepting the incorrectly padded values) onto all implementations.
But yes, this particular padding check is more appropriate in libssl than libcrypto.